java-checks/src/main/java/org/sonar/java/checks/TooManyParametersCheck.java

```diff
@@ -106,4 +137,16 @@ private static boolean usesAuthorizedAnnotation(MethodTree method) {
     return hasUnknownAnnotation(metadata) || WHITE_LIST.stream().anyMatch(metadata::isAnnotatedWith);
   }
 
+  //As of Spring 4.3, classes (@Component, @Service, etc..) with a single constructor can omit the @Autowired annotation.
+  private static boolean classUsesAuthorizedAnnotation(ClassTree methodParentClass) {
+    SymbolMetadata parentClassMetadata = methodParentClass.symbol().metadata();
+    //if the parent class is a Spring component or has unknown annotations
+    if(hasUnknownAnnotation(parentClassMetadata) || CLASS_CONSTRUCTOR_WHITELIST.stream().anyMatch(parentClassMetadata::isAnnotatedWith)) {  
+      long numberOfConstructors = methodParentClass.members().stream().filter(member -> member.is(Tree.Kind.CONSTRUCTOR)).count();    
+      //if it only has 1 constructor, @Autowired could be implicit, and it's an exception to the rule
+      if(numberOfConstructors == 1) return true;
```

Comment:
It seems you don't have the proper formatting settings in your IDE, or you don't format on `save`.
I was expecting comments with a leading space `// if` and `if` with `{ }`

Reply:
No I don't have any automatic formatting on ctrl+s, I will see if I can get to set that up